src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+import { useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import './Autocomplete.scss';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [focusOnInput, setFocusOnInput] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [query, setQuery] = useState('');
+
+  const applyQuery = debounce(setAppliedQuery, delay);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+    applyQuery(value);
```

Comment:
You are not checking if the user entered spaces only. This can lead to unnecessary function calls. You should add a condition to check if the `value` is not empty or doesn't contain spaces only before calling `applyQuery(value)`.